### DISCUSSION TITLE
App-based auth and 2-factor

### DISCUSSION THREAD

**Discussion by TekExplorer** [pelican-dev/panel#120] [Ideas]
Hey all, I'm the developer of Featherdactyl (possibly to be renamed :/)

I wanted to start up a discussion on what it would take to set up easier authentication features in the app

1. How to log into the app directly, perhaps with a callback that provides some kind of auth token
2. How to include a steam guard-like 2 factor feature
   - perhaps a duo-like one-tap?
3. How to include a steam guard/discord-like qr-code login, no username/password required
4. App as passkey?

I believe all of these require some panel modifications, so its necessary to discuss it here.

additionally, if there is an existing flow we can just add, it could make it easier.

**Comment by rmartinoscar** [comment]
Resolved in https://github.com/pelican-dev/panel/issues/385
